### ISSUE TITLE
Unexpected limits when combining quiver with other plots using MB

### ISSUE THREAD

**Issue by Chrillebon** [dtudk/dtumathtools#26]
## Short description:
When using the matplotlib backend, the code listed in **Code block A** does not give the expected result (**plot B**). This is due to an interaction between forcing xlims/ylims vs autoscale on x/y in different plots being combined.

## Long description
When using the quiver function from matplotlib (2D example: `fig, ax = plt.subplots(1,1);ax.quiver(0.0, -2, 2, -1.0, angles= 'xy', scale_units= 'xy', scale= 1)`, and 3D example: `ax = plt.figure().add_subplot(projection='3d');ax.quiver(0.0, -2, 2, -1.0, 1.0, 1.0)`), the x and y (auto) limits only focus on the starting point of the quiver. Therefore, to get "automatic limits" on the vectors in dtuplot (2D example: `dtuplot.quiver(0.0, -2.0, 2.0, -1.0)`, and 3D example: `dtuplot.quiver(0.0, -2.0, 2.0, -1.0, 1.0, 1.0)`), we cannot rely on the auto-limits from matplotlib. The current workaround utilizes a detail of sympy-plot-backends, where the limits can be auto-generated from the data if the `_sal` flag is set, seen [here](https://github.com/Davide-sd/sympy-plot-backends/blob/9a2be84ec3e0aacf5443b5971af6acc5f57e1e0b/spb/backends/matplotlib/renderers/renderer.py#L45).

_Note 1: Because this is specific to the quiver in matplotlib, the behavior is not seen in the other backends_. 
_Note 2: The current `dtuplot.quiver()` function does not seem to perfectly align the limits of the plot with the vector. This issue has been [located and adressed](https://github.com/Davide-sd/sympy-plot-backends/pull/30), as [this line](https://github.com/Davide-sd/sympy-plot-backends/blob/9a2be84ec3e0aacf5443b5971af6acc5f57e1e0b/spb/backends/matplotlib/matplotlib.py#L502C33-L502C33) was missing from spb (version <3.0, see changelog [here](https://sympy-plot-backends.readthedocs.io/en/latest/changelog.html#v3-0-0)). Therefore this will be fixed when bumping spb to a higher version. This will not be done currently, as described [here](https://github.com/dtudk/dtumathtools/pull/24#issue-2072297350)._ 
_Note 3: The workaround forces the internal data structure of dtuplot-quivers to use "starting point" and "ending point" such that the correct limits are inferred from "the data"._

The effect of the workaround is that xlimits and ylimits are saved in the specific renderer for quiver, overwriting the autoscale that would have otherwise been used (section of code is found [here](https://github.com/Davide-sd/sympy-plot-backends/blob/9a2be84ec3e0aacf5443b5971af6acc5f57e1e0b/spb/backends/matplotlib/matplotlib.py#L485)). This has the intended effect when this/multiple vectors are plotted, but when combined with different plots (relying on the auto-scale), the actual xlims/ylims of the combined plot are not as would be expected (see **plot A** vs **plot B**). 

To generate the correct plot, one can then either remove the `_sal` flag (has to be done before plot is shown using `show=True` or `.show()`), as shown in **Code block B**, or simply (and much more intuitively) set the limits manually, as shown in **Code block C**.

## Resolving the issue?
A temporary solution would be to simply set the limits manually (**Code block C**). I cannot see any simple fix that solves this problem without affecting other plots using quivers/vectors. Might have to add a keyword argument where one can disable the "automatic limits" enforced by the quiver (removing the `_sal` flag), ensuring the automatic limits of matplotlib. Note that this would create another problem where quivers with ends outside the automatic limits of all other plots would be cropped. 

Another option would be to change over to the newly introduced [sympy-plot-backend versions of quiver](https://github.com/Davide-sd/sympy-plot-backends/pull/30#issuecomment-1729080465) with documentation [here](https://sympy-plot-backends.readthedocs.io/en/latest/modules/graphics/vectors.html#spb.graphics.vectors.arrow_2d). However, as 3D vectors do not yet exist here, this is not an optimal solution either.

### Code block A (reproduce error)
```
from dtumathtools import *
from sympy import *
init_printing()

u, t = symbols('u t', real=True)
r = Matrix([u*cos(u), u*sin(u)])
rd = diff(r,u)
u0 = 3*pi/2
rdu0 = rd.subs(u,u0)
ru0 = r.subs(u,u0)
T = ru0 + t*rdu0

p1 = dtuplot.plot_parametric(r[0], r[1],(u,0,4*pi),rendering_kw={"color":"red"},use_cm=False,show=False)
p2 = dtuplot.plot_parametric(T[0],T[1],(t,-1.5,1.5),rendering_kw={"color":"royalblue"},use_cm=False,show=False)
p3 = dtuplot.scatter(ru0,rendering_kw={"markersize":10,"color":'black'}, show=False)
p4 = dtuplot.quiver(ru0,rdu0,rendering_kw={"color":"black"},show=False)
p = p1+p2+p3+p4
p.show()
```
### Plot A (plot generated from Code block A)
![image](https://github.com/dtudk/dtumathtools/assets/32595016/e9b90bd3-9269-47bd-9b72-9bfccc85ae67)

### Code block B (In addition to code block A, this additional line gives the desired result)
```
p = p1+p2+p3+p4
p._renderers[-1]._sal = False
p.show()
```
### Plot B (plot generated from both code blocks above)
![image](https://github.com/dtudk/dtumathtools/assets/32595016/f3620efc-afce-48c0-9ac9-0252aecc448b)

### Code block C (A more intuitive workaround until fixed, also giving plot B)
```
p = p1+p2+p3+p4
p.xlim = [-10,13]
p.ylim = [-12,8]
p.show()
```

**Comment by zerothi** [comment]
I'll let you decide what the best approach is here. Should we do anything before next release?

**Comment by Chrillebon** [comment]
I think the most unintrusive solution would be the option to turn off (by keyword argument) the forced limits by quiver, defaulting to auto-scale. I am unsure if I will have time to look at it before the next release, but if I do, it will come in a PR next week. 

Otherwise, are there anything that we are missing before next release?

**Comment by zerothi** [comment]
Ok, lets do that. And then a release this week.

**Comment by zerothi** [comment]
Once this is in, I will branch off and create 2024.1.

**Comment by zerothi** [comment]
Fixed in #27
